Mathlib/MeasureTheory/Constructions/EventuallyMeasurable.lean

```diff
@@ -0,0 +1,116 @@
+/-
+Copyright (c) 2024 Felix Weilacher. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Felix Weilacher
+-/
+
+import Mathlib.MeasureTheory.MeasurableSpace.Defs
+import Mathlib.Order.Filter.CountableInter
+
+/-!
+# Measurability modulo a filter
+
+In this file we consider the general notion of measurability modulo a σ-filter.
+Two important instances of this construction are null-measurability with respect to a measure,
+where the filter is the collection of co-null sets, and
+Baire-measurability with respect to a topology,
+where the filter is the collection of comeager (residual) sets.
+
+## Main definitions
+
+* `EventuallyMeasurableSpace`: A `MeasurableSpace` on a type `α` consisting of sets which are
+  `Filter.EventuallyEq` to a measurable set with respect to a given `CountableInterFilter` on `α`
+  and `MeasurableSpace` on `α`.
+* `EventuallyMeasurableSet`: A `Prop` for sets which are measurable with respect to some
+  `EventuallyMeasurableSpace`.
+* `EventuallyMeasurable`: A `Prop` for functions which are measurable with respect to some
+  `EventuallyMeasurableSpace` on the domain.
+
+-/
+
+open Filter Set MeasurableSpace
+
+variable {α : Type*} (m : MeasurableSpace α) (l : Filter α) [CountableInterFilter l] {s t: Set α}
```

Comment:
As you are making `m` and `l` implicit a few lines down, I think that perhaps it might be more readable to explicitly include them in `EventuallyMeasurableSpace` and `EventuallyMeasurableSet`.

Reply:
I prefer to declare `l` here since every declaration in this file depends on `[CountableInterFilter l]`.